app/models/annotations/status.rb

```diff
@@ -92,8 +93,21 @@ def update_elasticsearch_status
 
   def status_is_valid
     if !self.annotated_type.blank?
-      values = Status.possible_values(self.annotated, self.context)
+      annotated, context = get_annotated_and_context
+      values = Status.possible_values(annotated, context)
       errors.add(:base, 'Status not valid') unless values[:statuses].collect{ |s| s[:id] }.include?(self.status)
     end
   end
+
+  def get_annotated_and_context
+    if self.annotated_type == 'ProjectMedia' || self.annotated_type == 'ProjectSource'
+      annotated = self.annotated.media if self.annotated.respond_to?(:media)
+      annotated = self.annotated.source if obj.nil? and self.annotated.respond_to?(:source)
+      context = self.annotated.project if self.annotated.respond_to?(:project)
```

Comment:
`ProjectMedia` and `ProjectSource` context will be project otherwise it'll be `self.context`
also next step is to remove `context` from annotations